### DISCUSSION TITLE
Service role on SSR

### DISCUSSION THREAD

**Discussion by ntkris** [supabase/supabase#9425] [Questions]
Is it safe to use the service role on a server side request in NextJS?

```
export const getServerSideProps: GetServerSideProps = async (ctx) => {

  // Setup up supabase using my service key
  const serviceSupabase = getServiceSupabase()
  // Table with RLS
  const {data, error} = await serviceSupabase.from("table").select(*)


  return {
    props: { data: data },
  };
};

```

**Comment by GaryAustin1** [comment]
Service role key is fine to use for SSR as long as the key itself is never present on the browser client side.  So your getServiceSupabase call ,which I assumes sets up the url and key, must run only on the server and the key must be secure in how it is setup in env (ie not NEXT_PUBLIC_....).

**Reply by GaryAustin1** [reply]
Answering below follow up...
As long as getServiceSupabase function runs on the server (the service role key is only on the server).

**Comment by ntkris** [comment]
Thank you for answering, I only call by "getServiceSupabase" on SSR or the API routes. 

I'm assuming that's okay?

Thanks for your help
